R/visualization.R

```diff
@@ -0,0 +1,222 @@
+                               sd = sd(responses$response)))
+
+  write.table(all_resp %>% arrange(desc(.data$N)) %>%
+    select(.data$first_drug, .data$N, .data$response, .data$p, .data$purch_age_dist),
+    paste0(out_file_prefix, "_responses_by_drug.txt"), sep="\t", row.names=FALSE, quote=FALSE)
+
+
+  plot(ggtexttable(responses %>% group_by(.data$first_drug) %>%
+    summarise(n_purch=n(), n_indiv=length(unique(.data$FINNGENID)),
+     p=get_response_p_value(.data$response),
+     response=mean(.data$response),
+     purch_age_dist=quant_text(.data$baseline_age)) %>%
+     select(.data$first_drug, .data$n_purch, .data$response, .data$p, .data$purch_age_dist) %>%
+    arrange(desc(.data$n_purch))))
+
+
+  begin <- ceiling(max(min(-labs$time_to_drug, na.rm = TRUE), drug_response$lab_response_period$before_period[1]))
+  end <- ceiling(min(max(-labs$time_to_drug, na.rm = TRUE), drug_response$lab_response_period$after_period[2]))
+  labs$bin <- cut(-labs$time_to_drug,
+                  breaks=seq(begin, end, by=.25),
+                  include.lowest = TRUE)
+
+  plot(ggplot(labs %>% filter(!is.na(.data$bin))) + geom_boxplot(aes(x=.data$bin, y=.data$MEASUREMENT_VALUE_HARMONIZED)) +
+      labs(x="Time to drug purchase (years)", y="Lab measurement") +
+      ggtitle("Lab measurements before and after drug purchase")) +  theme_bw() +
+      theme(axis.text.x = element_text(angle = 45, size=20)) +
+
+
+  write.table(labs %>% group_by(.data$bin) %>%
+    summarise(n=n(), mean=mean(.data$MEASUREMENT_VALUE_HARMONIZED), sd=sd(.data$MEASUREMENT_VALUE_HARMONIZED)),
+    paste0(out_file_prefix, "_labs_by_time_to_drug.txt"), sep="\t", row.names=FALSE, quote=FALSE)
+
+
+  plot(ggplot(responses) + geom_histogram(aes(x=.data$response)) + theme_bw() +
+      labs(x="Response (after - before)", y="Count") +
+      ggtitle("Distribution of drug response"))
+
+  fit <- summary(lm(after ~ before, data=responses))
+
+  slope <- format(fit$coefficients["before","Estimate"], digits=2)
+  r2 <- format(fit$r.squared, digits=2)
+  p <- format(fit$coefficients["before","Pr(>|t|)"], digits=2, scientific=TRUE)
+
+  suppressWarnings(print(ggplot(responses, aes(x=.data$before, y=.data$after)) + geom_point() + geom_smooth(method="lm") +
+      geom_abline(slope=1) +
+      ggtitle(paste0("Before vs. after values. Slope: ", slope, " R2: ", r2, " p: ", p))))
+
+  dev.off()
+
+  print(paste0("Created summary plots and tables with prefix: ", out_file_prefix))
+
+}
+
+
+#' @title Plot Distribution of Lab Values Before and After Drug Use
+#' @description Creates a boxplot comparing the distribution of lab values
+#' before and after the first drug purchase, faceted by drug type.
+#' @param drug_response A `drug.response` object.
+#' @param remove_outliers A logical indicating whether to remove outliers
+#' using the 1.5 * IQR rule. Defaults to `FALSE`.
+#' @return A `ggplot` object.
+#' @export
+plot_lab_value_distribution <- function(drug_response, remove_outliers = FALSE) {
```

Comment:
as discussed this plot is not necessarily very informative. The p-values are not valid as there are multiple values per individual before and after. 
At least remove the p-values. MAybe density or violin plot could be a bit more informative so that odd shapes like densities at certain values would be possible to spot